src/libmain-c/nix_api_main.cc

```diff
@@ -17,4 +18,14 @@ nix_err nix_init_plugins(nix_c_context * context)
     NIXC_CATCH_ERRS
 }
 
+nix_err nix_set_log_format(nix_c_context * context, const char * format)
+{
+    if (context)
+        context->last_err_code = NIX_OK;
+    try {
+        nix::setLogFormat(format);
```

Comment:
in c++ this is a reference so we need: https://github.com/DeterminateSystems/nix-src/pull/211#discussion_r2373329150

Reply:
Done